static/app/components/quickTrace/utils.tsx

```diff
@@ -1,86 +1,29 @@
+import type {Event, EventTransaction} from 'sentry/types/event';
 import type {Organization} from 'sentry/types/organization';
-import {defined} from 'sentry/utils';
 import EventView from 'sentry/utils/discover/eventView';
 import {SavedQueryDatasets} from 'sentry/utils/discover/types';
-import {
-  eventDetailsRouteWithEventView,
-  generateEventSlug,
-} from 'sentry/utils/discover/urls';
-import type {
-  EventLite,
-  QuickTraceEvent,
-  TraceError,
-  TracePerformanceIssue,
-} from 'sentry/utils/performance/quickTrace/types';
-import {getTraceTimeRangeFromEvent} from 'sentry/utils/performance/quickTrace/utils';
 import {hasDatasetSelector} from 'sentry/views/dashboards/utils';
 import type {TraceViewSources} from 'sentry/views/performance/newTraceDetails/traceHeader/breadcrumbs';
-import type {TraceTree} from 'sentry/views/performance/newTraceDetails/traceModels/traceTree';
 import {getTraceDetailsUrl} from 'sentry/views/performance/traceDetails/utils';
 
-export function isQuickTraceEvent(
-  event: QuickTraceEvent | TraceError | TracePerformanceIssue
-): event is QuickTraceEvent {
-  return defined((event as QuickTraceEvent)['transaction.duration']);
+function isTransaction(event: Event): event is EventTransaction {
+  return event.type === 'transaction';
 }
 
-export type ErrorDestination = 'discover' | 'issue';
-
-function generateIssueEventTarget(
-  event: TraceError | TracePerformanceIssue | TraceTree.EAPError,
-  organization: Organization,
-  referrer?: string
-): LocationDescriptor {
-  const queryParams = referrer ? '?referrer=' + referrer : '';
-  return `/organizations/${organization.slug}/issues/${event.issue_id}/events/${event.event_id}/${queryParams}`;
-}
-
-function generateDiscoverEventTarget(
-  event: EventLite | TraceError | TracePerformanceIssue,
-  organization: Organization,
-  location: Location,
-  referrer?: string
-): LocationDescriptor {
-  const eventSlug = generateEventSlug({
-    id: event.event_id,
-    project: event.project_slug,
-  });
-  const newLocation = {
-    ...location,
-    query: {
-      ...location.query,
-      project: String(event.project_id),
-      ...(referrer ? {referrer} : {}),
-    },
-  };
-  return eventDetailsRouteWithEventView({
-    organization,
-    eventSlug,
-    eventView: EventView.fromLocation(newLocation),
-    isHomepage: location.query.homepage === 'true' || undefined,
-  });
-}
-
-export function generateSingleErrorTarget(
-  event: TraceError | TracePerformanceIssue,
-  organization: Organization,
-  location: Location,
-  destination: ErrorDestination,
-  referrer?: string
-): LocationDescriptor {
-  switch (destination) {
-    case 'issue':
-      return generateIssueEventTarget(event, organization, referrer);
-    case 'discover':
-    default:
-      return generateDiscoverEventTarget(event, organization, location, referrer);
-  }
+export function getTraceTimeRangeFromEvent(event: Event): {end: string; start: string} {
```

Comment:
Moved utils from performance/quickTrace to this file. It contains some routing utils used widely in the app, so we are keeping this file around till the next milestone, which intends to improve the trace view routing api.